sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -354,181 +466,50 @@ export class BatchingReceiver extends MessageReceiver {
-        if (reuse) msg += " Receiver link already present, hence reusing it.";
-        log.batching(msg, this._context.namespace.connectionId, maxWaitTimeInMs, this.name);
-        totalWaitTimer = setTimeout(actionAfterWaitTimeout, maxWaitTimeInMs);
-      };
+        receiver.addCredit(args.maxMessageCount);
 
-      if (!this.isOpen()) {
-        log.batching(
-          "[%s] Receiver '%s', setting max concurrent calls to 0.",
-          this._context.namespace.connectionId,
-          this.name
-        );
-        // while creating the receiver link for batching receiver the max concurrent calls
-        // i.e. the credit_window on the link is set to zero. After the link is created
-        // successfully, we add credit which is the maxMessageCount specified by the user.
-        this.maxConcurrentCalls = 0;
-        const rcvrOptions = this._createReceiverOptions(false, {
-          onMessage: onReceiveMessage,
-          onError: onReceiveError,
-          onSessionError: onSessionError,
-          onSettled: onSettled,
-          onClose: onReceiveClose,
-          onSessionClose: onSessionClose
-        });
-        this._init(rcvrOptions, abortSignal)
-          .then(() => {
-            if (!this._receiver) {
-              // there's a really small window here where the receiver can be closed
-              // if that happens we'll just resolve to an empty array of messages.
-              return resolve([]);
-            }
-
-            // TODO: long-term we probably need to split the code in this promise. This check
-            // is just a band-aid for now.
-            if (!abortSignal?.aborted) {
-              this._receiver.on(ReceiverEvents.receiverDrained, onReceiveDrain);
-              addCreditAndSetTimer();
-            }
-            return;
-          })
-          .catch(reject);
-      } else {
-        addCreditAndSetTimer(true);
-        this._receiver!.on(ReceiverEvents.message, onReceiveMessage);
-        this._receiver!.on(ReceiverEvents.receiverError, onReceiveError);
-        this._receiver!.on(ReceiverEvents.receiverDrained, onReceiveDrain);
-        this._receiver!.session.on(SessionEvents.sessionError, onSessionError);
-      }
-    });
-  }
+        let msg: string = `${loggingPrefix} Setting the wait timer for ${args.maxWaitTimeInMs} milliseconds`;
+        if (reuse) msg += " Receiver link already present, hence reusing it.";
+        log.batching(msg);
 
-  private _getRemainingWaitTimeInMsFn(
-    maxWaitTimeInMs: number,
-    maxTimeAfterFirstMessageInMs: number
-  ): () => number {
-    return getRemainingWaitTimeInMsFn(maxWaitTimeInMs, maxTimeAfterFirstMessageInMs);
-  }
+        totalWaitTimer = setTimeout(actionAfterWaitTimeout, args.maxWaitTimeInMs);
+      };
 
-  private _getServiceBusMessage(context: EventContext): ServiceBusMessageImpl {
-    return new ServiceBusMessageImpl(this._context, context.message!, context.delivery!, true);
-  }
+      addCreditAndSetTimer(true);
```

Comment:
Previously having `addCreditAndSetTimer()` as a separate function was useful because it was called from two places. One was for when a new links was created and the other where the link was re-used. That's why the argument `reuse` which was used only for logging purposes.

With the refactoring in this PR, this method is called only from 1 place. I would recommend inlining its contents.

Reply:
Makes sense.

Also, I can get rid of the `reuse` check. :)